contracts/token/ERC721/ERC721SingleRentalAgreement.sol

```diff
@@ -47,16 +44,21 @@ contract ERC721SingleRentalAgreement is Context, IERC721RentalAgreement, ERC165
     event FundsRedeemed(address redeemer, uint256 amount, uint256 remainingBalance);
 
     constructor(
-        address _owner,
-        address _renter,
-        address _erc721Contract,
-        uint256 _duration,
-        uint256 _expirationDate,
+        IERC721Rental _erc721Contract,
+        uint256 _tokenId,
+        uint40 _duration,
+        uint40 _expirationDate,
         uint256 _rentalFees
     ) {
-        owner = _owner;
-        renter = _renter;
+        // Original owner.
+        if (_erc721Contract.rentedOwnerOf(_tokenId) != address(0)) {
```

Comment:
Maybe put the result in a variable to avoid doing 2 calls.

Reply:
+1 I changed it